packages/o-spreadsheet-engine/src/helpers/recompute_zones.ts

```diff
@@ -181,25 +181,31 @@ export function modifyProfiles<T extends Zone | UnboundedZone>( // export for te
 export function profilesContainsZone(
   profilesStartingPosition: number[],
   profiles: Map<number, number[]>,
-  zone: Zone
+  zone: Zone | UnboundedZone
 ): boolean {
   const leftValue = zone.left;
   const rightValue = zone.right;
-  const topValue = zone.top;
-  const bottomValue = zone.bottom + 1;
   const leftIndex = binaryPredecessorSearch(profilesStartingPosition, leftValue, 0);
-  const rightIndex = binaryPredecessorSearch(profilesStartingPosition, rightValue, leftIndex);
-  if (leftIndex === -1 || rightIndex === -1) {
-    return false;
-  }
+  const rightIndex =
+    rightValue === undefined
+      ? profilesStartingPosition.length - 1
+      : binaryPredecessorSearch(profilesStartingPosition, rightValue, leftIndex);
+  /**
+   * The `profilesStartingPosition` array always contains at least the value `0` at its first position,
+   * ensuring that applying `binaryPredecessorSearch` will always return a valid index.
+   * Therefore, it is not necessary to check if the result of `binaryPredecessorSearch` equals `-1`.
+   */
+  const topValue = zone.top;
+  const bottomValue = zone.bottom === undefined ? undefined : zone.bottom + 1;
   for (let i = leftIndex; i <= rightIndex; i++) {
     const profile = profiles.get(profilesStartingPosition[i])!;
-    const topPredIndex = binaryPredecessorSearch(profile, topValue, 0, true);
-    const bottomSuccIndex = binarySuccessorSearch(profile, bottomValue, 0, true);
+    const topPredIndex = binaryPredecessorSearch(profile, topValue, 0);
     if (topPredIndex === -1 || topPredIndex % 2 !== 0) {
       return false;
     }
-    if (topValue < profile[topPredIndex] || bottomValue > profile[bottomSuccIndex]) {
+    const bottomSuccIndex =
+      bottomValue === undefined ? profile.length : binarySuccessorSearch(profile, bottomValue, 0);
+    if (topPredIndex + 1 !== bottomSuccIndex) {
```

Comment:
Why did you change this line ? (Probably a stupid question but I don't understand :/ )

Reply:
Here, the two previous conditions (`topValue < profile[topPredIndex]` and `bottomValue > profile[bottomSuccIndex]`) were not sufficient to ensure that the area fell within the current profile. Because a profile can consist of subsections, and these conditions failed to pass the case where a zone straddled two subsections:
https://github.com/odoo/o-spreadsheet/pull/7377/files#diff-6706ec8ed56f19533f131a8c007021c40fdba10d6d9d1780313f2b3300e1a358R291

The new condition resolves this by ensuring that `topPredIndex` and `bottomSuxxIndex` are indeed registered within the same profile subsection (in other words, that the indexes are consecutive here).